pkg/abaputils/manageGitRepositoryUtils.go

```diff
@@ -0,0 +1,132 @@
+package abaputils
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"reflect"
+	"sort"
+	"time"
+
+	piperhttp "github.com/SAP/jenkins-library/pkg/http"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// PollEntity periodically polls the pull/import entity to get the status. Check if the import is still running
+func PollEntity(repositoryName string, connectionDetails ConnectionDetailsHTTP, client piperhttp.Sender, pollIntervall time.Duration) (string, error) {
+
+	log.Entry().Info("Start polling the status...")
+	var status string = "R"
+
+	for {
+		var resp, err = GetHTTPResponse("GET", connectionDetails, nil, client)
+		if err != nil {
+			err = HandleHTTPError(resp, err, "Could not pull the Repository / Software Component "+repositoryName, connectionDetails)
+			return "", err
+		}
+		defer resp.Body.Close()
+
+		// Parse response
+		var abapResp map[string]*json.RawMessage
+		var body PullEntity
+		bodyText, _ := ioutil.ReadAll(resp.Body)
+
+		json.Unmarshal(bodyText, &abapResp)
+		json.Unmarshal(*abapResp["d"], &body)
+
+		if reflect.DeepEqual(PullEntity{}, body) {
+			log.Entry().WithField("StatusCode", resp.Status).WithField("repositoryName", repositoryName).Error("Could not pull the Repository / Software Component")
+			var err = errors.New("Request to ABAP System not successful")
+			return "", err
+		}
+
+		status = body.Status
+		log.Entry().WithField("StatusCode", resp.Status).Info("Pull Status: " + body.StatusDescription)
+		if body.Status != "R" {
+			PrintLogs(body)
+			break
+		}
+		time.Sleep(pollIntervall)
+	}
+	return status, nil
+}
+
+// PrintLogs sorts and formats the received transport and execution log of an import
+func PrintLogs(entity PullEntity) {
+
```

Comment:
We could think about saving the logs to the file system instead of printing them to the stdout. But I guess this would be a topic for another PR.